ast/term.go

```diff
@@ -2392,6 +2407,59 @@ func filterObject(o Value, filter Value) (Value, error) {
 	}
 }
 
+// NOTE(philipc): The only way to get an ObjectKeyIterator should be
+// from an Object. This ensures that the iterator can have implementation-
+// specific details internally, with no contracts except to the very
+// limited interface.
+type ObjectKeysIterator interface {
```

Comment:
[nit] The comment should be `// ObjectKeysIterator yaddayadda", some linter keeps yelling at me so often I've internalized it :D